[RHSSO-1737] Sync with EAP 7.2.1 modules change. Update templates

iankko opened the pull request:
[RHSSO-1737] Sync with EAP 7.2.1 modules change. Update templates
    version to '7.3.1.GA' & regenerate the docs
    
    See relevant EAP-7.2.1 changes:
      https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/215/files
      https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/215/files#r269169610
    
    for module version bump details
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

Commit:
[RHSSO-1737] Sync with EAP 7.2.1 modules change. Update templates
version to '7.3.1.GA' & regenerate the docs

See relevant EAP-7.2.1 changes:
  https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/215/files
  https://github.com/jboss-container-images/jboss-eap-7-openshift-image/pull/215/files#r269169610

for module version bump details

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
@douglaspalmer @martin-kanis @luck3y PTAL (once got a chance)

Thank you!, Jan

luck3y commented on image.yaml:
```diff
@@ -55,11 +55,11 @@ modules:
             git:
                   url: https://github.com/jboss-container-images/jboss-eap-modules.git
-                  ref: EAP72-CR5
+                  ref: EAP_721
```
this should be sprint-31 : https://github.com/jboss-container-images/jboss-eap-7-openshift-image/blob/eap72-dev/cp-overrides.yaml#L12

Commit:
[RHSSO-1737] Move 'jboss-os' repository definition out from
'sso.config.launch.setup.73' module-level to image-level. Also
add image-level based definition of 'jboss-rhscl' repository

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
JFTR, the 2-nd commit in this PR does the following:

```
    [RHSSO-1737] Move 'jboss-os' repository definition out from
    'sso.config.launch.setup.73' module-level to image-level.
```
The image-level repository definition (instead of the module-level based one, the current state) was suggested by @goldmann in https://github.com/jboss-openshift/cct_module/pull/334#issue-261526895

```
    Also add image-level based definition of 'jboss-rhscl' repository
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>
```
The ```jboss-rhscl``` image-level repository definition is required post ```sprint-27``` [cct_module](https://github.com/jboss-openshift/cct_module) [EAP modules move](https://github.com/jboss-openshift/cct_module/commit/1ced90793d90dd162492614228a170ebe682e73e) to [jboss-eap-modules](https://github.com/jboss-container-images/jboss-eap-modules) repository.

Within this move, besides moving the modules, the repository definitions were also removed from module-level definitions & replaced with EAP image-level definition. Compare their [the original cct_module form](https://github.com/jboss-openshift/cct_module/blob/9c4b1cca98ecee975a24321f886129b35686006d/os-eap-db-drivers/module.yaml#L8) with the [jboss-eap-modules form](https://github.com/jboss-container-images/jboss-eap-modules/blob/master/os-eap-db-drivers/module.yaml#L8)

Without introducing the ``` jboss-rhscl``` repository back on / to Red Hat Single Sign-On for OpenShift image level, the local image builds now fail with the error like the following [*]:
```
...
2019-03-27 12:56:56,883 cekit        INFO     Docker: rsync-3.1.2-4.el7.x86_64
2019-03-27 12:56:57,159 cekit        ERROR    You can look inside the failed image by running 'docker run --rm -ti 4dc6a060acd0
 bash'
2019-03-27 12:56:57,159 cekit        ERROR    ('Image build failed, see logs above.', CekitError("Image build failed: 'The command '/bin/sh -c yum install -y rh-maven35 rh-mongodb32-mongo-java-driver postgresql-jdbc mysql-connector-java hostname python-enum34 python-requests openssl yum-utils unzip tar rsync     && yum clean all && rm -rf /var/cache/yum     && rpm -q rh-maven35 rh-mongodb32-mongo-java-driver postgresql-jdbc mysql-connector-java hostname python-enum34 python-requests openssl yum-utils unzip tar rsync' returned a non-zero code: 2'",))
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/cekit/cli.py", line 256, in run
    builder.build()
  File "/usr/lib/python2.7/site-packages/cekit/builders/docker_builder.py", line 98, in build
    raise CekitError(msg, ex)
CekitError: ('Image build failed, see logs above.', CekitError("Image build failed: 'The command '/bin/sh -c yum install -y rh-maven35 rh-mongodb32-mongo-java-driver postgresql-jdbc mysql-connector-java hostname python-enum34 python-requests openssl yum-utils unzip tar rsync     && yum clean all && rm -rf /var/cache/yum     && rpm -q rh-maven35 rh-mongodb32-mongo-java-driver postgresql-jdbc mysql-connector-java hostname python-enum34 python-requests openssl yum-utils unzip tar rsync' returned a non-zero code: 2'",))
```

while with the ```jboss-rhscl``` repository defined back, the image build succeeds again:
```
...
2019-03-27 15:20:57,599 cekit        INFO     Docker: Step 102/103 : WORKDIR /home/jboss
2019-03-27 15:20:57,853 cekit        INFO     Docker: ---> b3bc3c2d60f5
2019-03-27 15:20:57,855 cekit        INFO     Docker: Step 103/103 : CMD /opt/eap/bin/openshift-launch.sh
2019-03-27 15:20:57,950 cekit        INFO     Docker: ---> Running in b8eacfe85a44
2019-03-27 15:20:58,049 cekit        INFO     Docker: ---> 37a1466e6a7a
2019-03-27 15:20:58,067 cekit        INFO     Docker: Successfully built 37a1466e6a7a
2019-03-27 15:20:58,140 cekit        INFO     Image built and available under following tags: redhat-sso-7/sso73-openshift:1.0, redhat-sso-7/sso73-openshift:latest
2019-03-27 15:20:58,140 cekit        INFO     Finished!
```

and such image also runs correctly:
```
...
14:35:40,984 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 57) WFLYUT0021: Registered web context: '/auth' for server 'default-server'
14:35:41,013 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 53) WFLYSRV0010: Deployed "keycloak-server.war" (runtime-name : "keycloak-server.war")
14:35:41,135 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0212: Resuming server
14:35:41,143 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://127.0.0.1:9990/management
14:35:41,143 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0054: Admin console is not enabled
14:35:41,143 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: Red Hat Single Sign-On 7.3.1.GA (WildFly Core 6.0.12.Final-redhat-00001) started in 30002ms - Started 673 of 976 services (693 services are lazy, passive or on-demand)
```

P.S.: [*] This defect wouldn't / shouldn't exhibit by OSBS builds, since there is ```content_sets.yaml``` definition (ignoring the cekit's / concreate's injections of the repositories and using just the ones, defined in content set), but since it's better to be able to build the OpenShift RH-SSO image also locally, the image-level repository definitiions (the 2nd commit) is better to be applied.

goldmann commented on image.yaml:
```diff
@@ -97,6 +97,10 @@ modules:
           - name: os-logging
+packages:
+      repositories:
+          - jboss-os
```
This will no longer work in CEKit 3.0, FYI. Use content sets.
